### ISSUE TITLE
Randomly generated data as the data source

### ISSUE THREAD

**Issue by eulerreich** [nicholas-leonard/dp#160]
Currently, if I want to randomly generate sequences of bits as input and some transform of these sequences as output, I'd have to do this all at once before training, for a fixed number of samples, to make a data source. Is there a way to generate these data on the fly, as the model is requesting them?

**Comment by nicholas-leonard** [comment]
@eulerreich Yes, you could implement a DataSet subclass that does this for you. Just over-write the sub(), index() or sample() method. These work respectively with the Sampler, ShuffleSampler and RandomSampler samplers.
